middleware/classes.py

```diff
@@ -1,3 +1,6 @@
+# Third Party (PyPI) Imports
+from user_agents import parse
```

Comment:
style: Clear imports: Imports with relative names, obscure names, or generic names which have the potential to conflict with other symbols in a file should either just use simple module import and the fully-qualified function, or an `as` import to disambiguate and make clear.

Please either do:

`import user_agents` and use `user_agents.parse`

Or:

`from user_agents import parse as parse_user_agent`